zubhub_frontend/zubhub/src/App.js

```diff
@@ -457,32 +448,12 @@ function App(props) {
             )}
           />
 
-          <Route
-            path="/activities/:id/edit"
-            render={routeProps => (
-              <PageWrapper {...routeProps} {...props}>
-                <LazyImport LazyComponent={CreateActivity} {...routeProps} {...props} />
-              </PageWrapper>
-            )}
-          />
-
-          <Route
-            path="/activities/:id"
-            render={routeProps => (
-              <PageWrapper {...routeProps} {...props}>
-                <LazyImport LazyComponent={ActivityDetails} {...routeProps} {...props} />
-              </PageWrapper>
-            )}
-          />
-
-          <Route
-            path="/activities"
-            render={routeProps => (
-              <PageWrapper {...routeProps} {...props}>
-                <LazyImport LazyComponent={Activities} {...routeProps} {...props} />
-              </PageWrapper>
-            )}
-          />
+          <ProtectedRoute path="/activities/create" component={CreateEditActivity} {...props} />
+          <ProtectedRoute path="/activities/my-activities" component={MyActivities} {...props} />
+          <ProtectedRoute path="/activities/:id/linked-projects" component={LinkedProjects} {...props} />
+          <ProtectedRoute path="/activities/:id/edit" component={CreateEditActivity} {...props} />
+          <ProtectedRoute path="/activities/:id" component={ActivityDetails} {...props} />
+          <ProtectedRoute path="/activities" component={Activities} {...props} />
```

Comment:
Protecting all routes relate to `/activities`

Reply:
What is this `editting`?

Reply:
I talked about it [here](https://github.com/unstructuredstudio/zubhub/pull/885#discussion_r1357710524). Since am using the same component, `CreateEditActivity` component, when creating and editting an activity, I needed a way to switch between different label texts, titles, button text and also functionalities. 
Since editing only happens on the `/activity/:id/edit` route, I passed this boolean.

We can use the `url` with help of some regex, but thought this way is more precise and easier

Reply:
Ok, got it. Maybe we should do something about the typo, but fine for now 🤷